RealTimeFaceMaskDetector/TCPServer/SocketServer.cpp

```diff
@@ -58,37 +65,90 @@ bool SocketServer::BindListeningSocket()
 
 bool SocketServer::AcceptConnection()
 {
-	m_client_socket = accept(m_listen_socket, NULL, NULL);
-	if (m_client_socket == INVALID_SOCKET)
+	std::cout << "Waiting connection" << std::endl;
+	//Here make async wait
+	bool ready = false, stop = false;
+	SOCKET mock_socket = INVALID_SOCKET;
+	std::future<SOCKET> cl_socket = std::async(std::launch::async, [this,&ready] 
+		{ 
+			SOCKET result;
+
+			result = accept(m_listen_socket, NULL, NULL);
+
+			ready = true;
+			return result;
+		});
+	
+	std::cout << "Waiting...     Press \'ESC\' to stop the Server\n" << std::flush;
+
+	bool key = { false };
+	bool old_key = { false };
+	while (!ready)
 	{
+		//wait
+		key = GetAsyncKeyState(VK_ESCAPE) & 0x01;
+		if (key && !old_key)
+		{
+			server_is_up = false;
+
+			sockaddr_in clientService;
+			clientService.sin_family = AF_INET;
+			clientService.sin_addr.s_addr = inet_addr(m_ip.c_str());
+			clientService.sin_port = htons(std::stoi(m_port));
+			
+			mock_socket = socket(m_host_info->ai_family, m_host_info->ai_socktype, m_host_info->ai_protocol);
+			connect(mock_socket, (SOCKADDR*)&clientService, sizeof(clientService));
```

Comment:
connect function is used for client sockets. Why have you used this function in the server part? Is this feature of "asio" sockets?

Reply:
It was used to interrupt accept function